scripts/src/javatests/org/oppia/android/scripts/maven/GenerateLicenseTextsTest.kt

```diff
@@ -0,0 +1,259 @@
+  private val VALID_LINK = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+  private val INVALID_LINK = "https://fabric.io/terms"
+
+  private val outContent: ByteArrayOutputStream = ByteArrayOutputStream()
+  private val originalOut: PrintStream = System.out
+
+  @Rule
+  @JvmField
+  var tempFolder = TemporaryFolder()
+
+  @Before
+  fun setUp() {
+    tempFolder.newFolder("values")
+    tempFolder.newFolder("assets")
+    System.setOut(PrintStream(outContent))
+  }
+
+  @After
+  fun restoreStreams() {
+    System.setOut(originalOut)
+  }
+
+  @Test
+  fun testScript_noArguments_printsUsageStringAndThrowsException() {
+    val exception = assertThrows(Exception::class) { main(arrayOf()) }
+
+    assertThat(exception).hasMessageThat().contains(TOO_LESS_ARGS_FAILURE)
+    assertThat(outContent.toString()).contains("Usage:")
+  }
+
+  @Test
+  fun testScript_sixArguments_printsUsageStringAndThrowsException() {
+    val exception = assertThrows(Exception::class) {
+      main(
+        arrayOf(
+          "${tempFolder.root}/values",
+          "test_third_party_dependency_names.xml",
+          "test_third_party_dependenct_versions.xml",
+          "test_third_party_dependency_license_texts.xml",
+          "test_third_party_dependency_license_texts_array.xml",
+          "test_third_party_dependency_license_names_array.xml"
+        )
+      )
+    }
+
+    assertThat(exception).hasMessageThat().contains(TOO_LESS_ARGS_FAILURE)
+    assertThat(outContent.toString()).contains("Usage:")
+  }
+
+  @Test
+  fun testScript_dependencyListEmpty_failsWithNotUpToDateException() {
+    val mavenDependencyList = getMavenDependencyList()
+
+    val pbFile = tempFolder.newFile("assets/test_maven_dependencies.pb")
+    mavenDependencyList.writeTo(pbFile.outputStream())
+
+    val exception = assertThrows(Exception::class) { runScript() }
+    assertThat(exception).hasMessageThat().contains(MAVEN_DEPENDENCY_LIST_NOT_UP_TO_DATE_FAILURE)
+  }
+
+  @Test
+  fun testScript_licenseListEmpty_failsWithNotUpToDateException() {
+    val dependencyList = listOf<MavenDependency>(getMavenDependency("artifact:name"))
+    val mavenDependencyList = getMavenDependencyList(dependencyList)
+
+    val pbFile = tempFolder.newFile("assets/test_maven_dependencies.pb")
+    mavenDependencyList.writeTo(pbFile.outputStream())
+
+    val exception = assertThrows(Exception::class) { runScript() }
+    assertThat(exception).hasMessageThat().contains(MAVEN_DEPENDENCY_LIST_NOT_UP_TO_DATE_FAILURE)
+  }
+
+  @Test
+  fun testScript_primaryLinkTypeUnspecified_failsWithNotUpToDateException() {
+    val licenseList = listOf<License>(
+      getLicense(
+        licenseName = "license_0",
+        primaryLink = VALID_LINK
+      )
+    )
+    val dependencyList = listOf<MavenDependency>(
+      getMavenDependency(
+        artifactName = "artifact:name",
+        licenseList = licenseList
+      )
+    )
+    val mavenDependencyList = getMavenDependencyList(dependencyList)
```

Comment:
Here & elsewhere: please only use ``main()`` when testing the script. We should never punch holes in the private implementation of a component when testing it, and instead should only test it through its public API.

Reply:
Done.